src/com/zone5cloud/retrofit/core/OkHttpClientInterceptor_Authorization.java

```diff
@@ -180,13 +184,16 @@ public Response intercept(Chain chain) throws IOException {
 	/** 
 	 * If we have a refreshable token and it is expired or near expiry, do a token refresh 
 	 * @param chain - The intercepted chain
-	 * @param baseUrl - The base hose url from the request, used to issue the refresh request
+	 * @param originalRequestUrl - The base hose url from the request, used to issue the refresh request
 	 **/
-	private void refreshIfRequired(Chain chain, String baseUrl) {
+	//auto refresh needs to use this configured baseURL for constructing the auto refresh request,
+	// not the baseURL of the current incoming request.
+	private void refreshIfRequired(Chain chain, String originalRequestUrl) {
+		RequestBody body = null;
```

Comment:
It is not the originalRequestUrl that we want. It is the zone5BaseURL that we need to construct the refresh request.